### PR TITLE
Update "update po" action in po smart contract

### PR DESCRIPTION
**Pull request by shannynalayna**
Updates the "update_purchase_order" method to handle a list of versions. This also updates the method to correctly handle versions depending on the intended update state. This also updates the related tests.

### CODE REVIEW

**Review by dnewh — APPROVED** [review]
tested

**Review by CLEckhardt — DISMISSED** [review]
Tested

File: contracts/purchase_order/src/handler.rs [hunk 1]
```diff
@@ -463,120 +556,317 @@ fn update_purchase_order(
...
+                if vers.is_draft() {
+                    return Err(ApplyError::InvalidTransaction(format!(
+                        "Purchase order {} version {} is a draft, \
+                        cannot accept a draft version",
+                        purchase_order.uid(),
+                        vers.version_id(),
+                    )));
+                }
+                vers.into_builder()
+                    .with_workflow_state("accepted".to_string())
```
Comment (vaporos) [inline contracts/purchase_order/src/handler.rs:849]: The smart contract should not know about specific workflow state strings like "accepted".

**Review by vaporos — CHANGES_REQUESTED** [review]

File: contracts/purchase_order/src/handler.rs (hunk 1 above)
Comment (shannynalayna) [reply contracts/purchase_order/src/handler.rs]: Updated to return errors if the smart contract ends up in a place where it finds other versions (besides the one specified in the po's accepted vers num) that are in a state with the `accepted` constraint or if the po's specified accepted version is not in a state with the `accepted` constraint

**Review by CLEckhardt — APPROVED** [review]
Tested

**Review by dnewh — APPROVED** [review]

**Review by vaporos — APPROVED** [review]

**Pull request merged**